Mathlib/Analysis/Complex/CauchyIntegral.lean

```diff
@@ -3,10 +3,7 @@ Copyright (c) 2021 Yury Kudryashov. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Yury Kudryashov
 -/
-import Mathlib.Analysis.Analytic.Uniqueness
 import Mathlib.Analysis.Calculus.DiffContOnCl
-import Mathlib.Analysis.Calculus.DSlope
-import Mathlib.Analysis.Calculus.FDeriv.Analytic
 import Mathlib.Analysis.Complex.ReImTopology
 import Mathlib.Analysis.Real.Cardinality
 import Mathlib.MeasureTheory.Integral.CircleIntegral
```

Comment:
It seems quite surprising that removing these imports does not break anything, since you are not changing any of the existing code in the file (just adding more new stuff). This is not in any way a criticism of your PR; rather, it suggests that our "check for redundant imports" CI step doesn't seem to have been working optimally.

Reply:
I believe what's happening here is that there are two ways for an import to be redundant:

1. Its contents are not needed
2. Its is transitively imported via another file

and `lake exe shake` never attempts to remove imports which are redundant in the second sense (which these are). I further believe that part of the reason for this is that if we aggressively remove all such imports, then the set of imports becomes a bit more brittle to refactors.

One could perhaps make a case for removing these imports but I think we should just leave them as they are until / unless we decided to alter the practice for how to handle such redundant imports repo-wide.